### DISCUSSION TITLE
Multi-factor Authentication

### DISCUSSION THREAD

**Discussion by jgibbs-pdx** [Snowflake-Labs/terraform-provider-snowflake#701] [Q&A]
We are testing out using this provider, and we require everyone to have MFA on their account.

When we run terraform apply, we get around 10 MFA prompts that we have to click through quickly before it times out. We are assuming this is because the apply is threaded, so it has to do the MFA prompt per thread.

We don't need terraform to run super frequently or super quickly, so we were wondering if there is a way to run it on one connection? Or are we just thinking about this completely wrong?

**Comment by FriedCircuits** [comment]
@jgibbs-pdx There is a terraform parameter to set parallelization to 1. That might help you.

**Comment by csp33** [comment]
If you use rsa keys, you won't have to answer the MFA prompt.
Just add the [private_key_path](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs#private_key_path) to the provider config
